utils_nlp/dataset/dac.py

```diff
@@ -36,3 +45,145 @@ def load_pandas_df(local_cache_path=None, num_rows=None):
+                i.e. token ids, corresponding to the tokens in the input 
+                text data.
+            2. input_mask_all: Tensor. Each sublist contains the attention
+                mask of the input token id list, 1 for input tokens and 0 for
+                padded tokens, so that padded tokens are not attended to.
+            4. label_ids_all: Tensor, each sublist contains token labels of
+                a input sentence/paragraph, if labels is provided. If the
+                `labels` argument is not provided, it will not return this tensor.
+
+        test_dataset (TensorDataset): A TensorDataset containing the following four tensors.
+            1. input_ids_all: Tensor. Each sublist contains numerical values,
+                i.e. token ids, corresponding to the tokens in the input 
+                text data.
+            2. input_mask_all: Tensor. Each sublist contains the attention
+                mask of the input token id list, 1 for input tokens and 0 for
+                padded tokens, so that padded tokens are not attended to.
+            4. label_ids_all: Tensor, each sublist contains token labels of
+                a input sentence/paragraph, if labels is provided. If the
+                `labels` argument is not provided, it will not return this tensor.
+        
+        label_encoder (LabelEncoder): a sklearn LabelEncoder instance. The label values
+            can be retrieved by calling the `inverse_transform` function.
+    """
+
+     # download and load the original dataset
+    all_df = load_pandas_df(
+        local_cache_path=local_path,
+        num_rows=None
+    )
+
+    # set the text and label columns
+    text_col = all_df.columns[0]
+    label_col = all_df.columns[1]
+
+    label_encoder = LabelEncoder()
+    label_encoder.fit(["culture", "diverse", "economy", "politics", "sports"])
+
+    # remove empty documents
+    all_df = all_df[all_df[text_col].isna() == False]
+
+    if test_fraction < 0 or test_fraction >= 1.0:
+        logging.warning("Invalid test fraction value: {}, changed to 0.25".format(test_fraction))
+        test_fraction = 0.25
+    
+    train_df, test_df = train_test_split(
+        all_df,
+        train_size=(1.0 - test_fraction),
+        random_state=random_seed
+    )
+
+    if train_sample_ratio > 1.0:
+        train_sample_ratio = 1.0
+        logging.warning("Setting the training sample ratio to 1.0")
+    elif train_sample_ratio < 0:
+        logging.error("Invalid training sample ration: {}".format(train_sample_ratio))
+        raise ValueError("Invalid training sample ration: {}".format(train_sample_ratio))
+    
+    if test_sample_ratio > 1.0:
+        test_sample_ratio = 1.0
+        logging.warning("Setting the testing sample ratio to 1.0")
+    elif test_sample_ratio < 0:
+        logging.error("Invalid testing sample ration: {}".format(test_sample_ratio))
+        raise ValueError("Invalid testing sample ration: {}".format(test_sample_ratio))
```

Comment:
maybe you can do DRY (Don't Repeat Yourself) here. This code also appears in bbc hindi

Reply:
It's not exactly the same. All 3 TC datasets have different raw file formats and need to process a little bit differently. But I agree most of the code are the same. I think it's better to add separate functions in 3 datasets to make it much clearly, such that we know those datasets are independent.

Reply:
this code:
```
    if train_sample_ratio > 1.0:
        train_sample_ratio = 1.0
        logging.warning("Setting the training sample ratio to 1.0")
    elif train_sample_ratio < 0:
        logging.error("Invalid training sample ration: {}".format(train_sample_ratio))
        raise ValueError("Invalid training sample ration: {}".format(train_sample_ratio))
    
    if test_sample_ratio > 1.0:
        test_sample_ratio = 1.0
        logging.warning("Setting the testing sample ratio to 1.0")
    elif test_sample_ratio < 0:
        logging.error("Invalid testing sample ration: {}".format(test_sample_ratio))
        raise ValueError("Invalid testing sample ration: {}".format(test_sample_ratio))
```
is the same in [multinli](https://github.com/microsoft/nlp-recipes/blob/e7d6bed4d5b7cae659e23725424fba14fa9f094a/utils_nlp/dataset/multinli.py), [dac](https://github.com/microsoft/nlp-recipes/blob/e7d6bed4d5b7cae659e23725424fba14fa9f094a/utils_nlp/dataset/dac.py) and [bbc_hindi](https://github.com/microsoft/nlp-recipes/blob/e7d6bed4d5b7cae659e23725424fba14fa9f094a/utils_nlp/dataset/bbc_hindi.py).

In situations like this in other repos what we do is to have a common file in dataset, maybe called dataset_utils.py or something like that, and refactor the code there. Additionally, a nice pattern coming from Abhinav in our team: https://github.com/Microsoft/Recommenders/wiki/Coding-Guidelines#if-our-code-is-going-to-fail-let-it-fails-fast. It means that is there is a check like the limits you are setting, it should be at the beginning of the function.

Reply:
I totally agree that we should put those common functions in a separate file. But in this case it would an overkill in my opinion. I saw the same coding pattern of the `load_pandas_df` function in those files. And I can't see too much gain for putting those lines in a separate file. Just my 2 cents.

Reply:
cool

Reply:
ok I see your point